### DISCUSSION TITLE
Model type variation still accepts all scalar fields

### DISCUSSION THREAD

**Discussion by marcosdiasdev** [prisma/prisma#8411] [Q&A] [answered]
I've followed the [docs](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types) and built a custom model type for accessing only a subset of my User model's fields. 

My User schema looks something like this:

```ts
model User {
  id        Int      @id @default(autoincrement())
  firstName String
  lastName  String
  email     String   @unique
  phone     String
  username  String   @unique
  password  String
}
```

My custom model type looks like this:
```ts
import { Prisma } from '@prisma/client';

const userPersonalData = Prisma.validator<Prisma.UserArgs>()({
  select: {
    firstName: true,
    lastName: true,
    email: true,
    phone: true,
    username: true,
  },
});

export type UserPersonalData = Prisma.UserGetPayload<typeof userPersonalData>;
```

And I'm trying to use this custom model to make my `update` method reject the `password` field (I'll have a password update logic in another method):

```ts
import { Injectable } from '@nestjs/common';
import { PrismaService } from 'src/prisma/prisma.service';
import { User, Prisma } from '@prisma/client';
import { UserPersonalData } from './user';

@Injectable()
export class UserService {
  constructor(private prismaService: PrismaService) {}

  async update(id, data: UserPersonalData): Promise<User> {
    return this.prismaService.user.update({
      where: { id },
      data,
    });
  }
}
```

The problem is my `update` method still accepts a `data` object with the `password` field. Wasn't it supposed to throw an Exception if I do as follow?

```ts
const userData = {
    firstName: 'John',
    lastName: 'Doe',
    password: '123456' // This shouldn't be here
  }

const userService = new UserService();
userService.update(1, userData);
```

I only get an Exception if I try to put a field that don't exist in the original User model.

I tried to add `password: false` in the `userPersonalData` declaration but didn't have any success.

Am I missing something?

**Comment by marcosdiasdev** [comment, marked as answer]
I just realized the type isn't checked in runtime! My bad.

**Reply by ryands17** [reply]
Yup. Specified in the docs here: https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/prisma-validator

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
